In [48]:
# Probability of winning at least one block

N_blocks = 100
invested_others = 5000
invested_me = 1

P = 1 - (invested_others/(invested_others+invested_me))**N_blocks

print(N_blocks*invested_me, P, "  total invested, probability of winning once")

# for i in range(1,N_blocks+1):
#     P = 1 - (invested_others/(invested_others+invested_me))**i
#     print(i,P)



100 0.01979936655528325   total invested, probability of winning once


So it is expensive to try to win blocks on your own.  With enough funds and spreading out over a lot of blocks, you should get a fairly predictable amount of City Coins.  But without that, i.e., with small numbers, it's just gambling.

So it makes sense to join a mining pool, which I did.

In [9]:
# Estimated return on winning and stacking CityCoins, in the very first cycle.
# This assumes perfect odds, where you get a share of total coins, proportional to what you pay in.
# Obviously if you win more or fewer coins than this, your return adjusts proportionally.
# It also assumes you immediately stack all of whatever you win.
# This calculates the return in terms of STX committed at the beginning.  The fluctuating price of STX 
# is an additional risk to this investment.

n_blockspercycle = 2100
frac_winnings_stacked = 0.8  # Fraction of total number of coins stacked by everyone

N_cycles = 24   # Number of stacking cycles, each lasting ~2 weeks.

cycle_win = 0   # Cycle you earn your winnings (and thus stack in the next cycle)

# This assumes the average amount spent by everyone per block is constant (very crude).

# average_block_cost = 5000  
# coins_per_block = 250000
# principle = 500  # Not needed because only relative returns are calculated.

tot_interest = 0
N_coins = 0

print("Interest is expressed as a fraction of the principle amount used to mine coins.")


# This loop is set up only to work for the first ~4 years (after which rewards decrease further).


for i in range(1,N_cycles+1):

    N_coins_incycle = 0
    for j in range(1,n_blockspercycle+1):
        if (i-1)*n_blockspercycle + j < 10000:   # Bonus blocks
            coins_per_block = 250000 
        else:                                    # For next ~4 years
            coins_per_block = 100000
        N_coins_incycle += coins_per_block   # Total coins in cycle previous to stacking cycle
    N_coins += N_coins_incycle    # Total coins overall

#     print(N_coins_incycle, N_coins)

    N_coins_infirstcycle = n_blockspercycle * 250000    # Assumes rewards during bonus cycles only.
    new_interest = 0.7 * N_coins_infirstcycle / (frac_winnings_stacked*N_coins) 
    if i > cycle_win:
        tot_interest += new_interest
    print("cycle",i, "your cycle",i-cycle_win,"  Interest =",round(new_interest,3), " Total interest =",round(tot_interest,3))

Interest is expressed as a fraction of the principle amount used to mine coins.
cycle 1 your cycle 1   Interest = 0.875  Total interest = 0.875
cycle 2 your cycle 2   Interest = 0.438  Total interest = 1.312
cycle 3 your cycle 3   Interest = 0.292  Total interest = 1.604
cycle 4 your cycle 4   Interest = 0.219  Total interest = 1.823
cycle 5 your cycle 5   Interest = 0.18  Total interest = 2.003
cycle 6 your cycle 6   Interest = 0.166  Total interest = 2.17
cycle 7 your cycle 7   Interest = 0.155  Total interest = 2.324
cycle 8 your cycle 8   Interest = 0.144  Total interest = 2.469
cycle 9 your cycle 9   Interest = 0.136  Total interest = 2.604
cycle 10 your cycle 10   Interest = 0.128  Total interest = 2.732
cycle 11 your cycle 11   Interest = 0.121  Total interest = 2.852
cycle 12 your cycle 12   Interest = 0.114  Total interest = 2.967
cycle 13 your cycle 13   Interest = 0.109  Total interest = 3.075
cycle 14 your cycle 14   Interest = 0.103  Total interest = 3.179
cycle 15 your cy

The above just calculates the return on stacking coins.  But you still have the coins themselves, which might be worth nothing, but likely will trade for value.

The implied mining cost of a coin is just the average amount put up by all for each block, divided by the reward.  The reward is 250k coins in the first 10,000 blocks (around 2-3 months), then it goes down to 100k for the next ~4 years.  It then halves every ~4 years for around 20 years or so, before flattening out at 3125 coins per block forever.

So, if we ASSUME that the amount people put up for each block is approximately constant (a big assumption), then the implied mining cost increases by a factor of 2.5 after the bonus round, then doubles every 4 years.  Doubling every 4 years is like an annual growth of 20%.

Miami Coin is tradeable on Okcoin, and its price so far (after just \~3 months) has approximately tracked the implied mining cost.  They have \~1000 STX (\~\\$2k) per block, and the price fluctuated a bunch in the first couple of weeks and is now \\$0.02 (which is the mining cost for 100k coins won per block).

NYC Coin in cycle 0 has been doin an average of approximately 5000-6000 STX per block.  This implies a mining cost of \~\\$0.04 now and going up to \~\\$0.10 in a couple of months, when the reward goes to 100k coins per block.

We put in 1000 STX (\~\\$2300), so we expect to earn \~42000 NYC Coins.  So maybe the sale price of those coins could be \~\\$4200.